### DISCUSSION TITLE
Support more advanced message routing

### DISCUSSION THREAD

**Discussion by ryn9** [fluent/fluent-bit#5802] [Ideas]
A number of our fluentD configs/logic make use of labels and this has been sorely missed when using fluentbit.

Examples of how labels work can be seen here: https://docs.fluentd.org/configuration/routing-examples

While fluentbit's tag handling obviously is very powerful - conceptually labels can have a number of advantages.

Additionally - a concept like labels can can help folks whom have come from logstash and have used pipelines.

Perhaps labels, as implemented in fluentD, may not be exactly what your team would like to implement - but I think supporting more advanced message routing will help out with more complex configurations.

**Comment by ryn9** [comment]
Related to this would be how plugins could handle unrecoverable filters/output errors.
For example - unrecoverable errors for output. fluentD output plugins sometimes implement this with an `@ERROR` label, but there does not seem to be any plumbing for this type of situation in fluentbit.

**Comment by epsteina16** [comment]
This would be useful, and could be used in additions to tag as the primary form of routing. 
An error route, or retries failed route would be very useful in many situations. For example, if I don't want to lose any logs but my output is down, say elasticsearch, and my retry_limit of 5 is hit, it would be useful to be able to route the chunk that failed to write to a local log file.

**Comment by agup006** [comment]
Roadmap items we have been thinking that are applicable
* Output filters
* Secondary
